solr/core/src/java/org/apache/solr/cluster/Cluster.java

```diff
@@ -31,6 +31,7 @@
    */
   Set<Node> getLiveNodes();
 
+  Set<Node> getLiveNodes(boolean filterNonDataNodes) ;
```

Comment:
This feels like an overspecialization. Would it be better to have as an entirely separate method? `getLiveDataNodes()`?

Reply:
Fixed, thanks.